code/modules/vehicles/mecha/_mecha.dm

```diff
@@ -1082,32 +1033,34 @@
 			return
 		else
 			if(!AI.linked_core)
-				to_chat(AI, "<span class='userdanger'>Inactive core destroyed. Unable to return.</span>")
+				if(!silent)
+					to_chat(AI, "<span class='userdanger'>Inactive core destroyed. Unable to return.</span>")
 				AI.linked_core = null
 				return
-			to_chat(AI, "<span class='notice'>Returning to core...</span>")
+			if(!silent)
+				to_chat(AI, "<span class='notice'>Returning to core...</span>")
 			AI.controlled_mech = null
 			AI.remote_control = null
 			mob_container = AI
 			newloc = get_turf(AI.linked_core)
 			qdel(AI.linked_core)
 	else
 		return ..()
-	var/mob/living/L = M
+	var/mob/living/ejector = M
 	mecha_flags  &= ~SILICON_PILOT
-	if(mob_container.forceMove(newloc))//ejecting mob container
```

Comment:
?

Reply:
Please clarify what you mean here

Reply:
Why remove the check here? Was it redundant?

Reply:
Correct